presentation/src/main/java/com/threegap/bitnagil/presentation/splash/SplashViewModel.kt

```
@@ -46,34 +49,46 @@ class SplashViewModel @Inject constructor(
 
-    private fun checkTokenStatus() {
+    private fun performAutoLogin() {
         viewModelScope.launch {
             try {
-                hasToken = hasTokenUseCase()
-                sendIntent(SplashIntent.SetTokenChecked(hasToken))
+                val userRole = withTimeoutOrNull(5000) {
+                    autoLoginUseCase()
+                }
+                sendIntent(SplashIntent.SetUserRole(userRole))
             } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)